sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -426,31 +470,41 @@ cdef class Splitter:
                 # (min_samples_leaf, etc) and the grower will later turn the
                 # node into a leaf.
                 split_infos[feature_idx].gain = -1
-
-                # We will scan bins from left to right (in all cases), and if
-                # there are any missing values, we will also scan bins from
-                # right to left. This way, we can consider whichever case
-                # yields the best gain: either missing values go to the right
-                # (left to right scan) or to the left (right to left case).
-                # See algo 3 from the XGBoost paper
-                # https://arxiv.org/abs/1603.02754
-
-                self._find_best_bin_to_split_left_to_right(
-                    feature_idx, has_missing_values[feature_idx],
-                    histograms, n_samples, sum_gradients, sum_hessians,
-                    value, monotonic_cst[feature_idx],
-                    lower_bound, upper_bound, &split_infos[feature_idx])
-
-                if has_missing_values[feature_idx]:
-                    # We need to explore both directions to check whether
-                    # sending the nans to the left child would lead to a higher
-                    # gain
-                    self._find_best_bin_to_split_right_to_left(
-                        feature_idx, histograms, n_samples,
-                        sum_gradients, sum_hessians,
+                split_infos[feature_idx].is_categorical = is_categorical[feature_idx]
+
+                if is_categorical[feature_idx]:
+                    self._find_best_bin_to_split_category(
+                        feature_idx, has_missing_values[feature_idx],
+                        histograms, n_samples, sum_gradients, sum_hessians,
+                        value, monotonic_cst[feature_idx], lower_bound,
+                        upper_bound, &split_infos[feature_idx])
+                else:
+                    # We will scan bins from left to right (in all cases), and
+                    # if there are any missing values, we will also scan bins
+                    # from right to left. This way, we can consider whichever
+                    # case yields the best gain: either missing values go to
+                    # the right (left to right scan) or to the left (right to
+                    # left case). See algo 3 from the XGBoost paper
+                    # https://arxiv.org/abs/1603.02754
+                    # Note: for the categorical features above. this isn't
```

Comment:
@lorentzenchr Looks like I managed to remove your review comment, sorry

We don't need the capitalization here (also it's not user facing so let's not worry too much about this). I'll fix the dot vs comma